test/units/path_test.rb

```diff
@@ -54,6 +54,18 @@ def test_path_with_multiple_params
     assert_equal url, path.to_url
   end
 
+  def test_relative_path_with_params
+    url = "https://demo.imgix.net/images/demo.png?h=200&w=200&s=d570a1ecd765470f7b34a69b56718a7a"
+    path = client.path("images/demo.png").h(200).w(200)
+    assert_equal url, path.to_url
+  end
+
+  def test_file_path_with_reserved_delimiters
+    url = "https://demo.imgix.net/%20%3C%3E%5B%5D%7B%7D%7C%5C%5C%5E%25.jpg?h=200&w=200&s=c53e7dc75b2d8fb70006f12357881622"
```

Comment:
This is technically correct, but I wanted to point out that there are two `\` that are being encoded here instead of the original one `\` on the path we're using to test for resolution. If you were to append this to @sherwinski test source it would 404:

```
resolves: test_domain + "/%20%3C%3E%5B%5D%7B%7D%7C%5C%5E%25.jpg"
404s:      test_domain + "/%20%3C%3E%5B%5D%7B%7D%7C%5C%5C%5E%25.jpg"
```

We need to test that the paths we generate resolve and that's the only reason I'm bringing this up. The way I caught this was by appending it to the test source domain and noticed that it 404s (which hinted to me that maybe you didn't do a similar check, but I realize that if there were a test source that had two `\` in the path, this function would handle them appropriately).

You don't _have_ to change anything, but this is a reminder that we need to test for "does this URL resolve?"

Reply:
Thanks for the heads up! It's a typo carried over from having to double escape stuff in either csharp or java, can't recall now.